pages/domains/[name]/index.tsx

```diff
@@ -149,153 +142,37 @@ export function ManageNamePageContent({ name }: { name: string }) {
-                        <InfoCircleSVG className="text-gray-400 h-4 w-4 mr-1" />
-                      </div>
-                    </Skeleton> */}
-                    <div className="flex flex-col items-start justify-center gap-1">
-                      {ensData?.texts?.["email"] && (
-                        <Link
-                          target="_blank"
-                          href={`mailto:${ensData?.texts?.["email"]}`}
-                          className="p-2 flex gap-2 group"
-                        >
-                          <EmailIcon className="w-5 h-5 text-gray-400 group-hover:text-black transition-colors duration-200" />
-                          <h3 className="text-gray-400 group-hover:text-black transition-colors duration-300">
-                            {ensData?.texts?.["email"]}
-                          </h3>
-                        </Link>
-                      )}
-
-                      {!!ensData?.texts?.["com.github"] && (
-                        <Link
-                          target="_blank"
-                          href={`https://github.com/${ensData?.texts["com.github"]}`}
-                          className="p-2 flex gap-2 group"
-                        >
-                          <GithubIcon className="w-5 h-5 text-gray-400 group-hover:text-black transition-colors duration-200" />
-                          <h3 className="text-gray-400 group-hover:text-black transition-colors duration-300">
-                            {ensData?.texts?.["com.github"]}
-                          </h3>
-                        </Link>
-                      )}
-
-                      {ensData?.texts?.["com.twitter"] && (
-                        <Link
-                          target="_blank"
-                          href={`https://x.com/${ensData?.texts["com.twitter"]}`}
-                          className="p-2 flex gap-2 group"
-                        >
-                          <TwitterIcon className="w-5 h-5 text-gray-400 group-hover:text-black transition-colors duration-200" />
-                          <h3 className="text-gray-400 group-hover:text-black transition-colors duration-300">
-                            {ensData?.texts?.["com.twitter"]}
-                          </h3>
-                        </Link>
-                      )}
-
-                      {ensData?.texts?.["com.linkedin"] && (
-                        <Link
-                          target="_blank"
-                          href={`https://www.linkedin.com/in/${ensData?.texts["com.linkedin"]}`}
-                          className="p-2"
-                        >
-                          <LinkedInIcon className="w-5 h-5" />
-                        </Link>
-                      )}
-                    </div>
-                  </div>
-                </div>
+                <UserDomainCard
+                  name={name}
+                  avatar={ensData?.resolver.texts?.avatar}
```

Comment:
I don't like how multiple times we are making usage of possible `ensData` and possible resolver texts in this file.

I think we can achieve an optimized version of this file by defining early conditional for `ensData` existence + constants defining common used code (e.g. `const domainTextRecords = ensData.resolver.texts`) + a function to abstract repeated operations such as `const getDomainTextRecord = (recordKey: string): string | undefined` for these multiple social medias getters...

Goal here is to better write our code, avoiding repetitions and displaying code without _this-might-have-an-edge-case-i-need-to-deeper-debug_ scenarios for future developers reading (e.g. null-coalescing)

What do you think about this, @eduramme ?